staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go

```diff
@@ -471,31 +477,188 @@ func TestUpdateNodesInExternalLoadBalancer(t *testing.T) {
 				{Service: newService("s1", "888", v1.ServiceTypeLoadBalancer), Hosts: nodes},
 				{Service: newService("s3", "999", v1.ServiceTypeLoadBalancer), Hosts: nodes},
 			},
+			worker: 5,
 		},
 		{
-			// One service has an external load balancer and one is nil: one call.
+			desc: "One service has an external load balancer and one is nil: one call.",
 			services: []*v1.Service{
 				newService("s0", "234", v1.ServiceTypeLoadBalancer),
 				nil,
 			},
 			expectedUpdateCalls: []fakecloud.UpdateBalancerCall{
 				{Service: newService("s0", "234", v1.ServiceTypeLoadBalancer), Hosts: nodes},
 			},
+			worker: 6,
+		},
+		{
+			desc: "Four services have external load balancer with only 2 workers",
+			services: []*v1.Service{
+				newService("s0", "777", v1.ServiceTypeLoadBalancer),
+				newService("s1", "888", v1.ServiceTypeLoadBalancer),
+				newService("s3", "999", v1.ServiceTypeLoadBalancer),
+				newService("s4", "123", v1.ServiceTypeLoadBalancer),
+			},
+			expectedUpdateCalls: []fakecloud.UpdateBalancerCall{
+				{Service: newService("s0", "777", v1.ServiceTypeLoadBalancer), Hosts: nodes},
+				{Service: newService("s1", "888", v1.ServiceTypeLoadBalancer), Hosts: nodes},
+				{Service: newService("s3", "999", v1.ServiceTypeLoadBalancer), Hosts: nodes},
+				{Service: newService("s4", "123", v1.ServiceTypeLoadBalancer), Hosts: nodes},
+			},
+			worker: 2,
 		},
 	}
 	for _, item := range table {
 		controller, cloud, _ := newController()
 
+		nodeLister := controller.nodeLister.(*fakeNodeLister)
+		for _, node := range nodes {
+			nodeLister.cache = append(nodeLister.cache, node)
+		}
```

Comment:
instead of this, how about just simply doing:
```
controller.nodeLister = newFakeNodeLister(nodes)
```

(and letting newFakeNodeLister take list of nodes as params)
?

Reply:
Fixed.